tangelo/algorithms/variational/adapt_vqe_solver.py

```diff
@@ -285,12 +290,28 @@ def rank_pool(self, pool_commutators, circuit, backend, tolerance=1e-3):
                 pool_over = deflate_circuit.inverse() + circuit
                 f_dict, _ = backend.simulate(pool_over)
                 gradient[i] += self.deflation_coeff * grad * f_dict.get("0"*self.vqe_solver.ansatz.circuit.width, 0)
-        max_partial = max(gradient)
+
+        return gradient
+
+    def choose_operator(self, gradients, tolerance=1e-3):
+        """Choose next operator to add according to the ADAPT-VQE algorithm.
+
+        Args:
+            list of float: Operator gradients corresponding to self.pool_operators.
+            tolerance (float): Minimum value for gradient to be considered.
+
+        Returns:
+            list of int: Index (list of length=1) of the operator with the
+                highest gradient. If it is not bigger than tolerance, returns
+                an empty list.
+        """
+        sorted_op_indices = sorted(range(len(gradients)), key=lambda k: gradients[k])
```

Comment:
The previous implementation looked for the maximum value in the gradient and returned its index.
This new implementation sorts / ranks all entries in the gradient and retrieves the last, corresponding to the largest value in there, which is more complicated.

What motivated this choice ? Was it related to stability ?

Reply:
Yeah, mainly stability, and I think this implementation is easier to debug in general. I can revert back to the original if you desire.

Reply:
if you are willing to import numpy then this line can be replaced by.
```python3
sorted_op_indices = np.argsort(gradients)
```

Reply:
@AlexandreF-1qbit Then I think it's ok to keep your implementation. I let you decide whether or not to implement James suggestion.

Reply:
I think I prefer not importing numpy for this one-time purpose in this file.